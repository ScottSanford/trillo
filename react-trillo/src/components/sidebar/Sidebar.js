import React from 'react'
import './Sidebar.css'

const menu = [
  { icon: 'home', title: 'Hotel' },
  { icon: 'aircraft-take-off', title: 'Flight' },
  { icon: 'key', title: 'Car Rental' },
  { icon: 'map', title: 'Tours' },
]

export default function Sidebar() {

  const navList = menu.map(anItem => {

    const { icon, title } = anItem
    const isFirst = anItem.icon === 'home'
      ? 'side-nav__item--active'
      : ''

    const navItemClasses = `side-nav__item ${isFirst}`

    return (
      <li key={icon} className={navItemClasses}>
        <a href="#" className="side-nav__link">
          <svg className="side-nav__icon">
            <use xlinkHref={`img/sprite.svg#icon-${icon}`} />
          </svg>
          <span>{title}</span>
        </a>
      </li>
    )
  })

  return (
    <nav className="sidebar">
      <ul className="side-nav">
        {navList}
      </ul>

      <div className="legal">
        &copy; 2018 by Trillo. All rights reserved.
        </div>
    </nav>
  )
}