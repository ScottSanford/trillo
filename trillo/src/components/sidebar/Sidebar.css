/* Flex Item */
.sidebar {
    background-color: var(--color-grey-dark-1);
    display: flex;
    flex: 0 0 18%; /* no grow, no shrink, keep width at 18% */
    flex-direction: column;
    justify-content: space-between;
}

/* SIDE NAVIGATION */
.side-nav {
    font-size: 1.4rem;
    list-style: none;
    margin-top: 3.5rem;
}

.side-nav__item {
    position: relative;
}

.side-nav__item:not(:last-child) {
    margin-bottom: 0.5rem;
}

@media only screen and (max-width: 56.25em) {
    .side-nav {
        display: flex;
        margin-top: 0;
    }

    .side-nav__item {
        flex: 1;
    }

    .side-nav__item:not(:last-child) {
        margin: 0;
    }
}

.side-nav__item::before {
    background-color: var(--color-primary);
    content: '';
    height: 100%;
    left: 0;
    position: absolute;
    top: 0;
    transform: scaleY(0);
    transition: transform 0.2s, width 0.4s cubic-bezier(1, 0, 0, 1) 0.2s,
        background-color 0.1s;
    width: 3px;
}
.side-nav__item:hover::before,
.side-nav__item--active::before {
    transform: scaleY(
        1
    ); /* grows from the middle because of transform-origin: center */
    width: 100%;
}

.side-nav__item:active::before {
    background-color: var(--color-primary-light);
}

.side-nav__link:link,
.side-nav__link:visited {
    color: var(--color-grey-light-1);
    padding: 1.5rem 3rem;
    position: relative;
    text-decoration: none;
    text-transform: uppercase;
    z-index: 10; /* only works in that position */
    display: flex;
    align-items: center; /* vertically align items in this container */
}

@media only screen and (max-width: 56.25em) {
    .side-nav__link:visited {
        justify-content: center;
        padding: 2rem;
    }
}

.side-nav__icon {
    fill: currentColor;
    height: 1.75rem;
    margin-right: 2rem;
    width: 1.75rem;
}

@media only screen and (max-width: 37.5em) {
    .side-nav__icon {
        height: 2rem;
        margin-right: 0;
        width: 2rem;
    }

    .side-nav span {
        display: none;
    }
}

/* LEGAL TEXT */
.legal {
    color: var(--color-grey-light-4);
    font-size: 1.2rem;
    padding: 2.5rem;
    text-align: center;

    @media only screen and (max-width: 56.25em) {
        display: none;
    }
}