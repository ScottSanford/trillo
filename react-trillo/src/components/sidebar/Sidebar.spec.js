import React from 'react'
import { shallow } from 'enzyme'
import Sidebar from './Sidebar'

describe('Sidebar Component', () => {

	it('should match snapshot', () => {
		const wrapper = shallow(<Sidebar />)
		expect(wrapper).toMatchSnapshot()
	})

	it('should have 4 nav items', () => {
		const wrapper = shallow(<Sidebar />)
		expect(wrapper.find('.side-nav').children()).toHaveLength(4)
	})
})