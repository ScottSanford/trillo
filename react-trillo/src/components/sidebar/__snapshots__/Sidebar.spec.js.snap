// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component should match snapshot 1`] = `
<nav
  className="sidebar"
>
  <ul
    className="side-nav"
  >
    <NavItem
      item={
        Object {
          "icon": "home",
          "title": "Hotel",
        }
      }
      key="home"
    />
    <NavItem
      item={
        Object {
          "icon": "aircraft-take-off",
          "title": "Flight",
        }
      }
      key="aircraft-take-off"
    />
    <NavItem
      item={
        Object {
          "icon": "key",
          "title": "Car Rental",
        }
      }
      key="key"
    />
    <NavItem
      item={
        Object {
          "icon": "map",
          "title": "Tours",
        }
      }
      key="map"
    />
  </ul>
  <div
    className="legal"
  >
    Â© 2018 by Trillo. All rights reserved.
  </div>
</nav>
`;
